var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -855,25 +598,32 @@ def cmake_args(self):
         args = [
             self.define_from_variant("BUILD_SHARED_LIBS", "lib"),
             self.define_from_variant("LAMMPS_EXCEPTIONS", "exceptions"),
-            "-D{0}_MPI={1}".format(mpi_prefix, "ON" if "+mpi" in spec else "OFF"),
+            self.define_from_variant("{}_MPI".format(mpi_prefix), "mpi"),
             self.define_from_variant("BUILD_OMP", "openmp"),
-            "-DENABLE_TESTING=ON",
+            self.define("ENABLE_TESTING", self.run_tests),
         ]
-        if spec.satisfies("+cuda"):
-            args.append("-DPKG_GPU=ON")
-            args.append("-DGPU_API=cuda")
-            cuda_arch = spec.variants["cuda_arch"].value
-            if cuda_arch != "none":
-                args.append("-DGPU_ARCH=sm_{0}".format(cuda_arch[0]))
-            args.append(self.define_from_variant("CUDA_MPS_SUPPORT", "cuda_mps"))
-        elif spec.satisfies("+opencl"):
-            args.append("-DPKG_GPU=ON")
-            args.append("-DGPU_API=opencl")
-        else:
-            args.append("-DPKG_GPU=OFF")
+        if "~kokkos" in spec:
+            # LAMMPS can be build with the GPU package OR the KOKKOS package
```

Comment:
It feels like you are silently ignoring `+ cuda` when `+kokkos`, wouldn't it be better to add an explicit conflict?

Reply:
I didn't want to add a conflict yet. I still have much more GPU related changes incoming and don't want to be too restrictive. It'll build, it's just a weird mode of operation.

Reply:
Just to clarify, what I would want in the end is `+gpu+opencl`, `+gpu+cuda`, `+gpu+rocm` OR `+kokkos+cuda`, `+kokkos+rocm`. So if I would add a conflict, it would be between `+gpu` and `+kokkos`, even though it will probably even compile. It just is a source of troubles if you try to mix them in the same binary. But I'm thinking of moving the GPU related things into a new PR.